mail: upgrade mailserver fork

mkg20001 opened the pull request:
Note: Currently referencing my own fork, re-hosted on git.mkg since
github doesn't like the 100mb files of upstream and import failed last
week when I tried it

I've rebased all commits from christian kauhaus, you can review the list
here
https://git.mkg20001.io/mkg20001/fc-nixos-mailserver/-/commits/fc-rebase

@flyingcircusio/release-managers

## Release process

Impact:
- maybe mailservices will be restarted

Changelog:
- upgrade nixos-mailserver to latest master

## Security implications

- [x] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
  - upgrade nixos-mailserver
- [x] Security requirements tested? (EVIDENCE)
  - tested using tests and testvm

dpausp commented on versions.json:
```diff
@@ -6,9 +6,11 @@
-    "repo": "nixos-mailserver",
-    "rev": "d4566383b01c081bed866d397909a9e4fbefc7e2",
-    "sha256": "0hl31fjn0jf6j2rhf8g799q9vgmixyyy5j9srfa1njhy1npb04v3"
+    "url": "https://git.mkg20001.io/mkg20001/fc-nixos-mailserver",
```
change this to point to the fcio gitlab

dpausp commented on nixos/services/mail/default.nix:
```diff
@@ -156,7 +156,7 @@ in {
         enableImapSsl = true;
         enableManageSieve = true;
-        lmtpSaveToDetailMailbox = "no";
+        # lmtpSaveToDetailMailbox = "no";
```
can we just remove the line?

mkg20001 replied:
Option is removed by one of the nixos-mailserver fc commits, should we drop those we'd need to re-enable it, I think

But need to check again

mkg20001 replied:
Ah so option wasn't removed, this was just back when I pulled some random revision and had things break. We can leave it set to no.

mkg20001 committed:
mail: upgrade nixos-mailserver with customizations re-applied

This uses a fork which has the fc commits applied ontop of the
latest version of nixos-mailserver

fixes PL-129997

I've rebased all commits from christian kauhaus, you can review the list
here
https://git.mkg20001.io/mkg20001/fc-nixos-mailserver/-/commits/fc-rebase

dpausp commented:
@mkg20001 Change is ok, but the commit message is misleading now and needs a change
